lib/api/LogManagerImpl.cpp

```diff
@@ -807,12 +807,53 @@ namespace MAT_NS_BEGIN
     void LogManagerImpl::SetDataInspector(const std::shared_ptr<IDataInspector>& dataInspector)
     {
         LOCKGUARD(m_dataInspectorGuard);
-        m_dataInspector = dataInspector;
+        if(dataInspector == nullptr)
+        {
+            LOG_WARN("Attempting to set nullptr as DataInspector");
+            return;
+        }
+
+        auto itDataInspector = std::find_if(m_dataInspectors.begin(), m_dataInspectors.end(), [&dataInspector](const std::shared_ptr<IDataInspector>& currentInspector)
+        {
+            return strcmp(dataInspector->GetName(), currentInspector->GetName()) == 0;
+        });
+
+        if (itDataInspector != m_dataInspectors.end())
+        {
+            LOG_WARN("Replacing specified IDataInspector with passed in inspector");
+            m_dataInspectors.erase(itDataInspector);
+        }
+
+        m_dataInspectors.push_back(dataInspector);
+    }
+
+    void LogManagerImpl::ClearDataInspectors()
+    {
+        LOCKGUARD(m_dataInspectorGuard);
+        std::vector<std::shared_ptr<IDataInspector>>{}.swap(m_dataInspectors);
```

Comment:
This is backwards.
`m_dataInspectors.swap({});`

Reply:
http://www.cplusplus.com/reference/vector/vector/swap-free/

Reply:
Is it? Looking at the clear documentation it shows that `x` gets cleared and reallocated doing it the opposite way https://www.cplusplus.com/reference/vector/vector/clear/

Reply:
there is also this: https://github.com/microsoft/cpp_client_telemetry/blob/125818fcf7637b956f0454420d3c7be87b58ad80/lib/filter/EventFilterCollection.cpp#L44
current example in the code

Reply:
I believe using swap to clear is what threw me off. Yeah, your code works: https://godbolt.org/z/vh1Gxq